### PR TITLE
[#578] Sort pie chart segments to get consistent colors

### PR DESCRIPTION
**Pull request by gabemart**
#578 

By sorting the aggregated data before feeding it to vega, we increase the chances that pie charts will have more consistent segment colors across charts

- [ ] Update release notes
- [ ] Test plan | Unit test | Integration test
- [ ] Copyright header
- [ ] Code formatting
- [ ] Documentation

### CODE REVIEW

File: client/src/utilities/chart.js [hunk 1]
```diff
@@ -195,6 +198,7 @@ export function getPieData(visualisation, datasets) {
   const { datasetId, spec } = visualisation;
   const dataset = datasets[datasetId];
   const bucketIndex = getColumnIndex(dataset, spec.bucketColumn);
+  const bucketColumnType = dataset.get('columns').get(bucketIndex).get('type');
```
Comment (jonase) [inline client/src/utilities/chart.js]: Instead of chained `get`s you can also use `getIn`: dataset.getIn(['columns', bucketIndex, 'type']);

**Comment by jonase** [comment]
👍 lgtm

**Comment by mtwestra** [comment]
@gabe, tested by uploading the following dataset:
first_col, second_col, third_col
yes, no, yes
yes, no, no
no, no,	
no, yes, maybe
yes,  , yes
no, no, yes

So first and second column only containing 'yes/no', and third column containing 'maybe' as extra. Missing items in second and third column.

Expected behaviour: pie charts of first and second column should show same colours for yes and no. The third column will show different colours, as 'maybe' is the first alphabetically.

Actual results: 
screen shot 2017-03-07 at 11 36 54
screen shot 2017-03-07 at 11 37 06
screen shot 2017-03-07 at 11 37 20

I think the second image shows something unexpected: the No data is inserted as a colour in between yes and no. I think 'No data' should be put as the last colour always? The final image is consistent with this.

**Comment by gabemart** [comment]
@mtwestra The cause of this problem is that we were replacing empty values with the "empty value" string (`No data` currently) before sorting / applying the color indexes. To fix, we just need to compare against the empty value string rather than doing a null check

**Comment by gabemart** [comment]
Color results are as expected after changes in https://github.com/akvo/akvo-lumen/pull/641


image

image

image

**Comment by mtwestra** [comment]
@gabe, excellent

**Pull request merged**